templates/jenkins-override.epp

```diff
@@ -1,5 +1,12 @@
-<%| Hash[String, Struct[{value => Any}]] $environment |-%>
+<%- | Hash[String, Struct[{value => Any}]] $environment,
+      Optional[Hash[String, String]] $dropin_config = undef
+| -%>
 [Service]
 <% $environment.each |$key, $entry| { -%>
 Environment="<%= $key %>=<%= $entry['value'] %>"
 <% } -%>
+<% unless $dropin_config =~ Undef { -%>
+<% $dropin_config.each |$key, $value| { -%>
+<%= $key %>=<%= $value %>
+<% } -%>
+<% } -%>
```

Comment:
Then the `Undef` line becomes redundant
```suggestion
<% $dropin_config.each |$key, $value| { -%>
<%= $key %>=<%= $value %>
<% } -%>
```

Reply:
Sorry, I am kind of a Puppet newbie. Thanks for the recommendations.
I will take this approach.

Reply:
No worries, we all had to start somewhere :)